quiz/test_auto_publish.py

```diff
@@ -1,31 +1,35 @@
+import functools
 from datetime import datetime, timedelta
 
 from django.core.management import call_command
 from django.test import TestCase
 from django.utils import timezone
 
-from quiz.models import Question
+from quiz.test_helpers import create_questions
+
+
+def publish_and_refresh_scheduled_questions(questions):
```

Comment:
This is only ever called with a single question. Would it be simpler to make it take a single question rather than a collection?

Reply:
Yes, it would be simpler from the call site. But my reasoning was: 1) to support any number of questions if the need arises; 2) but mainly to match the function name: we could have scheduled multiple questions outside of this function, so passing it only one question doesn't necessary mean that we will be publishing only this question. The problem is that the function depends on the global state (we can't pass a list of questions to the `auto_publish` command, AFAIK), and even now, the name is a little misleading, since we publish only the `SCH` questions, but want to refresh every question from the passed iterable. So for the function with one parameter the correct name would be something like `publish_scheduled_questions_and_refresh_this_question`. For multiple parameters `publish_scheduled_questions_and_refresh_these_questions`. The latter has been roughly transformed to the current shorter version. Let me know if you still want to change the number of parameters or have a better name and I'll fix that.

Reply:
Right, I didn't think about that. I actually think the tests are easier to understand if we just do this then:

```python
def auto_publish_all_questions():
    call_command("auto_publish", "--skip-tweet")
```

```python
def test_publishes_scheduled_question_from_the_past(self):
    past = timezone.now() - timedelta(hours=1)
    [q] = create_questions(1, state="SCH", publish_time=past)
    auto_publish_all_questions()
    q.refresh_from_db()
    self.assertEqual("PUB", q.state)
```

What do you think?

Btw I've noticed all your comments in the other issues too, I just have a bit limited time. I'm getting to them! :)

Reply:
That fixes the name issue. The reason I haven't splitted those 2 actions is that it's easy to forget to call `q.refresh_from_db()` after seemingly unrelated to `q` function `auto_publish_all_questions()` and get a dangerous illusion of a passed test. Let's just leave this snippet as is. Anyway, `Question.objects.get()` is used in other places instead of `refresh_from_db()` and extracting a single `call_command()` doesn't gain much benefit.

Reply:
Ok, let's just leave this.